Source/Server/Packets/OrientationData.c

```diff
@@ -1,20 +1,28 @@
 #include <Server/Packets/ReceivePackets.h>
 #include <Server/Server.h>
+#include <Util/Checks/VectorChecks.h>
 #include <Util/Physics.h>
 #include <math.h>
 
 void receive_orientation_data(server_t* server, player_t* player, stream_t* data)
 {
     (void) server;
-    float x, y, z;
-    x                 = stream_read_f(data);
-    y                 = stream_read_f(data);
-    z                 = stream_read_f(data);
+
+    float x = stream_read_f(data);
+    float y = stream_read_f(data);
+    float z = stream_read_f(data);
+
     if ((x + y + z) == 0) {
         return;
     }
+
+    if (!valid_vec3f((vector3f_t){x, y, z})) {
```

Comment:
Instead of dynamic conversion create a vector and convert the packet to vector usage.